runtime/str.go

```diff
@@ -711,8 +755,6 @@ func adjustIndex(i, l int) int {
 		i = 0
 	case i < 0:
 		i += l
-	case i > l:
-		i = l
```

Comment:
I think this check is crucial. If we don't adjust i in this case and we use it to slice a string, Go will panic: `panic: runtime error: slice bounds out of range`. The intended postcondition of this function is that i is valid for slicing a string of length l.

Is there another way to fix the bug you found? Perhaps we need to remove this function and do the logic inline so that we can handle that case?